static/js/services/search.js

```diff
@@ -65,9 +77,36 @@ var _ = require('underscore'),
           return $q.resolve([]);
         }
 
-        return _search(type, filters, options)
-          .then(function(results) {
-            return GetDataRecords(results, options);
+        return _search(type, filters, options, extensions)
+          .then(function(searchResults) {
+            var dataRecordsPromise = GetDataRecords(searchResults, options);
+
+            var result;
+            if (extensions.displayLastVisitedDate) {
+              var lastVisitedDatePromise = getLastVisitedDates(searchResults);
+
+              result = $q.all({
+                dataRecords: dataRecordsPromise,
+                lastVisitedDates: lastVisitedDatePromise
+              }).then(function(r) {
+                r.lastVisitedDates.forEach(function(dateResult) {
+                  var relevantDataRecord = r.dataRecords.find(function(dataRecord) {
+                    return dataRecord._id === dateResult.key;
+                  });
+
+                  if (relevantDataRecord) {
+                    relevantDataRecord.lastVisitedDate = dateResult.value.max;
+                    relevantDataRecord.sortByLastVisitedDate = extensions.sortByLastVisitedDate;
```

Comment:
It seems unnecessary to add this to every item in the list. Instead can the sorting algorithm check if `lastVisitedDate` is defined and assume that means it should use it to sort?

Reply:
We can't do that, because it is always defined if you have this permission so it can be displayed, regardless of sort order.

The other option is to refactor livelist to allow for the concept of switching sorting in a more high level way, but I am still working under the assumption that these things we are building will be removed after the IN if they do not prove useful, so I am trying to not rock the boat too much.